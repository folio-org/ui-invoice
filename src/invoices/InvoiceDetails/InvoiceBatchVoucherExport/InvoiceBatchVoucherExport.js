import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import {
  Col,
  KeyValue,
  Row,
} from '@folio/stripes/components';
import {
  FolioFormattedTime,
} from '@folio/stripes-acq-components';

import { BATCH_VOUCHER_EXPORT_STATUS_LABEL } from '../../../common/constants';
import ExportVoucherButton from '../../../settings/BatchGroupConfigurationSettings/ExportVoucherButton';
import BatchGroupValue from '../BatchGroupValue';

const InvoiceBatchVoucherExport = ({ batchVoucherExport, exportFormat }) => {
  return (
    <Row>
      <Col xs={4}>
        <BatchGroupValue
          id={batchVoucherExport.batchGroupId}
          label={<FormattedMessage id="ui-invoice.invoice.details.batchVoucherExport.batchGroup" />}
        />
      </Col>

      <Col xs={4}>
        <KeyValue
          label={<FormattedMessage id="ui-invoice.invoice.details.batchVoucherExport.fileName" />}
        >
          <FolioFormattedTime dateString={batchVoucherExport.end} />
          <ExportVoucherButton
            batchVoucherId={batchVoucherExport.batchVoucherId}
            fileName={batchVoucherExport.end}
            format={exportFormat}
          />
        </KeyValue>
      </Col>

      <Col xs={4}>
        <KeyValue
          label={<FormattedMessage id="ui-invoice.invoice.details.batchVoucherExport.status" />}
          value={<FormattedMessage id={BATCH_VOUCHER_EXPORT_STATUS_LABEL[batchVoucherExport.status]} />}
        />
      </Col>
    </Row>
  );
};

InvoiceBatchVoucherExport.propTypes = {
  batchVoucherExport: PropTypes.object,
  exportFormat: PropTypes.string,
};

export default InvoiceBatchVoucherExport;
