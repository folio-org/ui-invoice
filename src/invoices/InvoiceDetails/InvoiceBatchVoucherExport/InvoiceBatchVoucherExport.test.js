import { render } from '@folio/jest-config-stripes/testing-library/react';

import { batchVoucherExport } from '../../../../test/jest/fixtures';

import InvoiceBatchVoucherExport from './InvoiceBatchVoucherExport';

jest.mock(
  '../../VoucherExport/ExportVoucherButton/ExportVoucherButton',
  () => jest.fn(() => 'ExportVoucherButton'),
);
jest.mock('../BatchGroupValue', () => jest.fn(() => 'BatchGroupValue'));

const defaultProps = {
  batchVoucherExport,
  exportFormat: 'json',
};
const renderInvoiceBatchVoucherExport = (props = defaultProps) => render(
  <InvoiceBatchVoucherExport {...props} />,
);

describe('InvoiceBatchVoucherExport', () => {
  beforeEach(() => {
    global.document.createRange = global.document.originalCreateRange;
  });

  afterEach(() => {
    global.document.createRange = global.document.mockCreateRange;
  });

  it('should render correct structure for invoice batch voucher export', () => {
    const { asFragment } = renderInvoiceBatchVoucherExport();

    expect(asFragment()).toMatchSnapshot();
  });
});
