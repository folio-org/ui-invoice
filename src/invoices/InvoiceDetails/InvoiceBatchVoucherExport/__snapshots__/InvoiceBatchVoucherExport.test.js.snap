// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvoiceBatchVoucherExport should render correct structure for invoice batch voucher export 1`] = `
<DocumentFragment>
  <div
    class="row"
  >
    <div
      class="col-xs-4"
    >
      BatchGroupValue
    </div>
    <div
      class="col-xs-4"
    >
      <div
        class="kvRoot"
      >
        <div
          class="kvLabel"
        >
          ui-invoice.invoice.details.batchVoucherExport.fileName
        </div>
        <div
          class="kvValue"
          data-test-kv-value="true"
        >
          voucher naExportVoucherButton
        </div>
      </div>
    </div>
    <div
      class="col-xs-4"
    >
      <div
        class="kvRoot"
      >
        <div
          class="kvLabel"
        >
          ui-invoice.invoice.details.batchVoucherExport.status
        </div>
        <div
          class="kvValue"
          data-test-kv-value="true"
        >
          ui-invoice.settings.BatchVoucherExports.status.generated
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
